tests/5.1/metadirective/test_metadirective_nothing.c

```diff
@@ -0,0 +1,109 @@
+//===--------------------- test_metadirective_nothing.c ---------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+// 
+// Test for nothing directive within metadirectives. Runs a variety of
+// metadirectives that check if the nothing directive is properly rendered.
+// Primarily tests based on the fact that no matter what 'when' clause is 
+// rendered it should result in nothing, and thus no additional pragma should
+// be created. Thus, the threads should remain unchanged through this process
+// and the compiler should handle it properly. Handles both device and host
+// based tests.
+//
+////===---------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int metadirectiveOnDevice() {
+   int errors = 0;
+   int A[N];
+
+   for (int i = 0; i < N; i++) {
+      A[i] = 0;
+   }
+
+   #pragma omp metadirective \
+      when( device={kind(nohost)}: nothing ) \
+      when( device={arch("nvptx")}: nothing) \
+      when( implementation={vendor(amd)}: nothing ) \
+      default( target parallel map(tofrom: A) )
```

Comment:
In this case is the expectation that the default is never reached ? If so, the current context selectors don't cover all cases and is not future-safe. Since we are still on the host it will pass the the when in 31 and if it is not NVIDIA or AMD it will go to default. 
If we want to make sure that it NEVER reaches default, we could enclose the metadirective inside a target region (and remove "target" from default directive variant.

Reply:
looks good.